x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_enrichment/processors/grok/grok_ai_suggestions.tsx

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import {
+  EuiBadge,
+  EuiButton,
+  EuiButtonIcon,
+  EuiCallOut,
+  EuiCodeBlock,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiText,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useWatch, useFormContext } from 'react-hook-form';
+import { useAbortController } from '@kbn/observability-utils-browser/hooks/use_abort_controller';
+import { IngestStreamGetResponse, SampleDocument } from '@kbn/streams-schema';
+import type { FindActionResult } from '@kbn/actions-plugin/server';
+import { UseGenAIConnectorsResult } from '@kbn/observability-ai-assistant-plugin/public/hooks/use_genai_connectors';
+import { useBoolean } from '@kbn/react-hooks';
+import { useKibana } from '../../../../hooks/use_kibana';
+import { GrokFormState, ProcessorFormState } from '../../types';
+import { UseProcessingSimulatorReturn } from '../../hooks/use_processing_simulator';
+import { useStreamsEnrichmentContext } from '../../enrichment_context';
+
+const RefreshButton = ({
+  generatePatterns,
+  connectors,
+  selectConnector,
+  currentConnector,
+  isLoading,
+}: {
+  generatePatterns: () => void;
+  selectConnector?: UseGenAIConnectorsResult['selectConnector'];
+  connectors?: FindActionResult[];
+  currentConnector?: string;
+  isLoading: boolean;
+}) => {
+  const [isPopoverOpen, { off: closePopover, toggle: togglePopover }] = useBoolean(false);
+  const splitButtonPopoverId = useGeneratedHtmlId({
+    prefix: 'splitButtonPopover',
+  });
+
+  const onButtonClick = () => {
+    togglePopover(!isPopoverOpen);
+  };
```

Comment:
Switching to useBoolean meant we don't need to re-declare this function, but you can directly pass the `togglePopover` handler to the onClick event. This can be removed